extensions/roc-plugin-repo/src/commands/release.js

```diff
@@ -170,85 +192,190 @@ export default projects => ({
+                        { silent: true },
+                      );
+                    }),
+                  ).then(async () => {
+                    if (collectedRelease) {
+                      const releaseTag = await getTag(collectedRelease);
+                      ctx.releaseTag = releaseTag;
+                      return execute(`git tag ${releaseTag}`, { silent: true });
+                    }
+
+                    return Promise.resolve();
+                  });
+                }
+
+                selectedToBeReleased[0].tag = `v${selectedToBeReleased[0]
+                  .packageJSON.version}`;
+                return execute(
+                  `git tag v${selectedToBeReleased[0].packageJSON.version}`,
+                  {
+                    silent: true,
+                  },
+                );
+              },
+            },
           ]),
       },
       {
         title: 'Publishing to npm',
         skip: () => !git || !publish,
-        task: () => {
-          const newTasks = Object.keys(status)
-            .map(projectName => {
-              if (status[projectName].increment > versions.NOTHING) {
-                return {
-                  title: `${projectName}@${status[projectName].version}`,
-                  task: () => {
-                    let registry = '';
-                    const publishConfig =
-                      status[projectName].packageJSON.publishConfig;
-                    if (publishConfig && publishConfig.registry) {
-                      registry = `--registry='${publishConfig.registry}'`;
-                    }
+        task: () =>
+          new Listr(
+            selectedToBeReleased.map(project => ({
+              title: `${project.name}@${project.packageJSON.version}`,
+              task: () => {
+                let registry = '';
+                const publishConfig = project.packageJSON.publishConfig;
+                if (publishConfig && publishConfig.registry) {
+                  registry = `--registry='${publishConfig.registry}'`;
+                }
 
-                    const publishCommand = `npm publish ${registry} --tag ${tag}`;
+                const publishCommand = `npm publish ${registry} --tag ${distTag}`;
 
-                    return execute(publishCommand, {
-                      cwd: status[projectName].path,
-                    });
-                  },
-                };
-              }
-              return undefined;
-            })
-            .filter(Boolean);
-          return new Listr(newTasks);
-        },
+                return execute(publishCommand, {
+                  cwd: project.path,
+                });
+              },
+            })),
+          ),
       },
       {
         title: 'Pushing to remote',
-        skip: () => !git || !push,
+        skip: () => !git || !tag || !push,
         task: () =>
-          execute(`git push`, {
+          execute(`git push${tag ? ' && git push --tags' : ''}`, {
```

Comment:
Have you considered splitting pushing tags into a separate task/step? I do not think conditional operators look good inside string templates.